### PR TITLE
feat(#60): allow login via openID tokens

### PR DESCRIPTION
**Pull request by Nightapes**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -18,17 +18,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
```
Comment (kaklakariada) [inline .github/workflows/go.yml:21]: 👍

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+all: build
+
+install_deps:
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.45.2
+
+lint:
+	golangci-lint run --print-issued-lines=false --fix ./...
```
Comment (kaklakariada) [inline Makefile:7]: Does this also fix issues? Then better rename the task to `lint-fix` and create a separate task just for linting

File: connection.go [hunk 3]
```diff
@@ -281,14 +282,68 @@ func (c *connection) close(ctx context.Context) error {
...
+		authRequest.Username = c.config.user
+		authRequest.Password = password
+	}
+
+	if osUser, err := user.Current(); err == nil && osUser != nil {
+		authRequest.ClientOsUsername = osUser.Username
+	} else {
+		logCouldNotGetOsUser(err)
+	}
+	log.Printf("Login via authRequest, %v", authRequest)
```
Comment (kaklakariada) [inline connection.go]: Does this also log passwords and tokens?

File: connection.go [hunk 4]
```diff
@@ -281,14 +282,68 @@ func (c *connection) close(ctx context.Context) error {
...
+		ClientVersion:  "(unknown version)",
+		ClientRuntime:  runtime.Version(),
+		Attributes: Attributes{
+			Autocommit:         boolToPtr(c.config.autocommit),
+			CurrentSchema:      c.config.schema,
+			CompressionEnabled: boolToPtr(hasCompression),
+		},
+	}
+
+	if c.config.accessToken != "" {
```
Comment (kaklakariada) [inline connection.go:299]: Is it possible to extract this `if` into a separate method that returns the auth request and an error?

File: dsn_test.go [hunk 5]
```diff
@@ -189,3 +189,35 @@ func (suite *DriverTestSuite) TestConfigToDsnWithDefaultValues() {
 	config := NewConfig("sys", "exasol")
 	suite.Equal("exa:localhost:8563;user=sys;password=exasol", config.String())
 }
+
+func (suite *DriverTestSuite) TestConfigWithAccessToken() {
+	config := NewConfigWithAccessToken("TOKEN.JWT.TEST")
+	suite.Equal("exa:localhost:8563;accesstoken=TOKEN.JWT.TEST", config.String())
+}
```
Comment (kaklakariada) [inline dsn_test.go:196]: Add a test for tokens with a `;`

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: connection.go (hunk 3 above)
Comment (Nightapes) [reply connection.go]: Is already removed

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**